app/tasks/raster_tile_cache_assets/raster_tile_cache_assets.py

```diff
@@ -204,9 +206,15 @@ async def raster_tile_cache_validator(
     Used in asset route. If validation fails, it will raise an
     HTTPException visible to user.
     """
-    source_asset: ORMAsset = await get_asset(
-        input_data["creation_options"]["source_asset_id"]
-    )
+    try:
+        source_asset: ORMAsset = await get_asset(
+            input_data["creation_options"]["source_asset_id"]
+        )
+    except RecordNotFoundError as e:
+        raise HTTPException(status_code=404, detail=str(e))
+    except DataError as e:
+        raise HTTPException(status_code=400, detail=str(e))
+    
```

Comment:
What causes a DataError? Should we log it too?

Reply:
DataError is if the asset uuid isn't even the right format (e.g. too short).  The detail does have a correct useful message, and I'm returning a 400 error code because it's a client input mistake.